File: app/src/main/java/com/checkin/app/checkin/Misc/UpdateAppActivity.java
package com.checkin.app.checkin.Misc;

import android.app.AlertDialog;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.checkin.app.checkin.R;
import com.checkin.app.checkin.Utility.Constants;

public class UpdateAppActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        new AlertDialog.Builder(UpdateAppActivity.this)
                .setTitle(R.string.app_old_version_dialog_title)
                .setMessage(R.string.app_old_version_dialog_message)
                .setPositiveButton("Ok", (dialogInterface, i) -> {
                    dialogInterface.dismiss();
                    startActivity(new Intent(Intent.ACTION_VIEW, Constants.INSTANCE.getPLAY_STORE_URI()));
                    finish();
                })
                .setNegativeButton("Cancel", ((dialogInterface, i) -> {
                    dialogInterface.dismiss();
                    finish();
                }))
                .show();
    }
}

File: app/src/main/java/com/checkin/app/checkin/misc/activities/UpdateAppActivity.kt
package com.checkin.app.checkin.misc.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import butterknife.BindView
import butterknife.ButterKnife
import com.checkin.app.checkin.R
import com.checkin.app.checkin.utility.Constants.PLAY_STORE_URI

class UpdateAppActivity : AppCompatActivity() {
    @BindView(R.id.container_blocking_network_error)
    internal lateinit var containerError: ViewGroup
    @BindView(R.id.container_blocking_network_loading)
    internal lateinit var containerLoading: ViewGroup
    @BindView(R.id.tv_blocking_error_message)
    internal lateinit var tvErrorMessage: TextView
    @BindView(R.id.btn_blocking_error_try_again)
    internal lateinit var btnPlayStore: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_blocking_network)
        ButterKnife.bind(this)

        containerLoading.visibility = View.GONE
        containerError.visibility = View.VISIBLE
        tvErrorMessage.text = "${getString(R.string.app_old_version_dialog_title)}\n${getString(R.string.app_old_version_dialog_message)}"
        btnPlayStore.setText(R.string.app_old_version_button_update)
        btnPlayStore.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, PLAY_STORE_URI))
            finish()
        }
    }
}